From 6ce82ed4eeb025f0e7a528b2da080f3b8dcb8485 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 26 Oct 2017 12:55:59 +0300
Subject: [PATCH 179/448] staging: fsl-dpaa2/eth: Type conversion in dpni-cmd.h

Make sure all structure fields passed to MC as command
parameters have __leX types instead of uX.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpni-cmd.h |   64 ++++++++++++------------
 1 files changed, 32 insertions(+), 32 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpni-cmd.h b/drivers/staging/fsl-dpaa2/ethernet/dpni-cmd.h
index dcb5a4c..a9ee25f 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpni-cmd.h
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpni-cmd.h
@@ -521,45 +521,45 @@ struct dpni_cmd_set_queue {
 #define DPNI_DISCARD_ON_MISS_SIZE	1
 
 struct dpni_cmd_set_qos_table {
-	u32 pad;
+	__le32 pad;
 	u8 default_tc;
 	/* only the LSB */
 	u8 discard_on_miss;
-	u16 pad1[21];
-	u64 key_cfg_iova;
+	__le16 pad1[21];
+	__le64 key_cfg_iova;
 };
 
 struct dpni_cmd_add_qos_entry {
-	u16 pad;
+	__le16 pad;
 	u8 tc_id;
 	u8 key_size;
-	u16 index;
-	u16 pad2;
-	u64 key_iova;
-	u64 mask_iova;
+	__le16 index;
+	__le16 pad2;
+	__le64 key_iova;
+	__le64 mask_iova;
 };
 
 struct dpni_cmd_remove_qos_entry {
 	u8 pad1[3];
 	u8 key_size;
-	u32 pad2;
-	u64 key_iova;
-	u64 mask_iova;
+	__le32 pad2;
+	__le64 key_iova;
+	__le64 mask_iova;
 };
 
 struct dpni_cmd_add_fs_entry {
 	/* cmd word 0 */
-	u16 options;
+	__le16 options;
 	u8 tc_id;
 	u8 key_size;
-	u16 index;
-	u16 flow_id;
+	__le16 index;
+	__le16 flow_id;
 	/* cmd word 1 */
-	u64 key_iova;
+	__le64 key_iova;
 	/* cmd word 2 */
-	u64 mask_iova;
+	__le64 mask_iova;
 	/* cmd word 3 */
-	u64 flc;
+	__le64 flc;
 };
 
 struct dpni_cmd_remove_fs_entry {
@@ -569,9 +569,9 @@ struct dpni_cmd_remove_fs_entry {
 	u8 key_size;
 	__le32 pad1;
 	/* cmd word 1 */
-	u64 key_iova;
+	__le64 key_iova;
 	/* cmd word 2 */
-	u64 mask_iova;
+	__le64 mask_iova;
 };
 
 struct dpni_cmd_set_taildrop {
@@ -620,18 +620,18 @@ struct dpni_cmd_set_congestion_notification {
 	u8 tc;
 	u8 pad[6];
 	/* cmd word 1 */
-	u32 dest_id;
-	u16 notification_mode;
+	__le32 dest_id;
+	__le16 notification_mode;
 	u8 dest_priority;
 	/* from LSB: dest_type: 4 units:2 */
 	u8 type_units;
 	/* cmd word 2 */
-	u64 message_iova;
+	__le64 message_iova;
 	/* cmd word 3 */
-	u64 message_ctx;
+	__le64 message_ctx;
 	/* cmd word 4 */
-	u32 threshold_entry;
-	u32 threshold_exit;
+	__le32 threshold_entry;
+	__le32 threshold_exit;
 };
 
 struct dpni_cmd_get_congestion_notification {
@@ -642,19 +642,19 @@ struct dpni_cmd_get_congestion_notification {
 
 struct dpni_rsp_get_congestion_notification {
 	/* cmd word 0 */
-	u64 pad;
+	__le64 pad;
 	/* cmd word 1 */
-	u32 dest_id;
-	u16 notification_mode;
+	__le32 dest_id;
+	__le16 notification_mode;
 	u8 dest_priority;
 	/* from LSB: dest_type: 4 units:2 */
 	u8 type_units;
 	/* cmd word 2 */
-	u64 message_iova;
+	__le64 message_iova;
 	/* cmd word 3 */
-	u64 message_ctx;
+	__le64 message_ctx;
 	/* cmd word 4 */
-	u32 threshold_entry;
-	u32 threshold_exit;
+	__le32 threshold_entry;
+	__le32 threshold_exit;
 };
 #endif /* _FSL_DPNI_CMD_H */
-- 
1.7.1

